File: app/src/main/java/info/horriblesubs/sher/ui/common/LoadingDialog.java
package info.horriblesubs.sher.ui.common;

import android.app.Dialog;
import android.content.Context;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;

import java.util.Random;

import info.horriblesubs.sher.AppMe;
import info.horriblesubs.sher.R;
import pl.droidsonroids.gif.GifImageView;

public class LoadingDialog extends Dialog {

    private final GifImageView imageView;

    public LoadingDialog(@NonNull Context context) {
        super(context, AppMe.instance.getAppTheme() ? R.style.AniDex_Dialog_Dark : R.style.AniDex_Dialog_Light);
        setContentView(R.layout.layout_dialog_loading);
        imageView = new GifImageView(context);
        setCancelable(false);
        setLoadingGif();
    }

    private void setLoadingGif() {
        if (new Random().nextInt(100) % 2 == 0) imageView.setImageResource(R.drawable.loading_1);
        else imageView.setImageResource(R.drawable.loading_2);
        ((FrameLayout) findViewById(R.id.frameLayout)).addView(imageView);
    }
}

File: app/src/main/java/info/horriblesubs/sher/dialog/LoadingDialog.kt
package info.horriblesubs.sher.dialog

import android.content.Context
import android.widget.FrameLayout
import info.horriblesubs.sher.R
import pl.droidsonroids.gif.GifDrawable
import pl.droidsonroids.gif.GifImageView
import java.util.*

class LoadingDialog(context: Context): BaseDialog(context, R.layout._dialog_a) {
    private fun setLoadingGif() {
        val name = context.resources
            .getStringArray(R.array.asset_names)[Random().nextInt(6)]?:"gif_01.gif"
        val drawable = GifDrawable(context.assets, name)
        val imageView = GifImageView(context)
        imageView.setImageDrawable(drawable)
        findViewById<FrameLayout>(R.id.frameLayout).addView(imageView)
    }
    init {
        setCancelable(false)
        setLoadingGif()
    }
}